PAPP-22703: Mark Salesforce as FIPS Compliant

shiyingtu opened the pull request:
- Mark as fips_complaint
- Upgrade min_phantom_version
- Fix semgrep tests

shiyingtu-p committed:
Run pre-commit hook on all files

shiyingtu-p committed:
Fix semgrep and lines too long failures

shiyingtu-p committed:
Mark Salesforce as FIPS Compliant in app json

shiyingtu-p committed:
Upgrade min_phantom_version to 5.1.0

shiyingtu-p committed:
Merge branch 'next' into fips

shiyingtu-p committed:
Run pre-commit hook on all files

shiyingtu-p committed:
Upgrade Django version to 4.0.1

- Background: Django 3.2.11, 3.2.12 will fail whitesource scan
because of high security issues
- Fix: fixed as suggested, upgrade to 4.0.1

paulnawat commented on requirements.txt:
```diff
@@ -1,3 +1,3 @@
-Django==3.2.5
 beautifulsoup4==4.9.1
+Django==4.0.1
```
`4.0.1` can only be used for py38 and above. As a result, we can still see `3.2.5` being used in py36 below (see `pip_dependencies` field below.

Suggestion: I find `3.2.12` to pass the ProdScan and work on py36+

Otherwise, we can use `4.0.1` on py39 and `3.2.12` on py36. I just though `3.2.12` looks simpler.

shiyingtu replied:
Make sense. Changed the Django version

shiyingtu-p committed:
Change Django version to 3.2.12

4.0+ only works for python3.9, choose 3.2.12 for django
working in both python3.6 and 3.9

shiyingtu-p committed:
Update wheels file
